metadata-ingestion/src/datahub/ingestion/transformer/add_dataset_tags.py

```diff
@@ -40,22 +40,64 @@ def create(cls, config_dict: dict, ctx: PipelineContext) -> "AddDatasetTags":
         config = AddDatasetTagsConfig.parse_obj(config_dict)
         return cls(config, ctx)
 
-    def transform_one(self, mce: MetadataChangeEventClass) -> MetadataChangeEventClass:
-        assert isinstance(mce.proposedSnapshot, DatasetSnapshotClass)
-        tags_to_add = self.config.get_tags_to_add(mce.proposedSnapshot)
-        if tags_to_add:
-            tags = builder.get_or_add_aspect(
-                mce,
-                GlobalTagsClass(
-                    tags=[],
-                ),
+    @staticmethod
+    def get_patch_global_tags_aspect(
```

Comment:
it feels like there's a ton of copy-paste across all of these transformers - is there an easy way to reduce that duplication?

Reply:
We need to implement a template method pattern to reduce this duplicate code. 
We can add a ticket in the backlog for the same.